Library/Homebrew/download_strategy.rb

```diff
@@ -282,6 +283,14 @@ def initialize(name, resource)
   end
 
   def fetch
+    if ENV["HOMEBREW_ARTIFACT_DOMAIN"]
+      if @url.start_with?(BottleSpecification::DEFAULT_DOMAIN)
+        @url.sub!(BottleSpecification::DEFAULT_DOMAIN, ENV["HOMEBREW_ARTIFACT_DOMAIN"])
+      else
+        @url.sub!(/^((ht|f)tps?:\/\/)?/, "#{ENV["HOMEBREW_ARTIFACT_DOMAIN"]}/#{Utils::Bottles::Bintray.repository(tap)}/")
+      end
```

Comment:
Considering all the subclasses, I wonder this probably would be better to put in `_fetch`.